plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -143,9 +151,138 @@ func (k *KafkaConsumer) Init() error {
 	}
 
 	k.config = cfg
+
+	k.compileTopicRegexps()
+
+	if len(k.regexps) == 0 {
+		// There are no regexp-matched topics
+		k.allWantedTopics = k.Topics
+	}
+
 	return nil
 }
 
+func (k *KafkaConsumer) createTopicScanner() error {
+	if k.topicClient == nil {
+		client, err := sarama.NewClient(k.Brokers, k.config)
+		if err != nil {
+			return err
+		}
+		k.topicClient = client
+	}
+	return nil
+}
+
+func (k *KafkaConsumer) compileTopicRegexps() {
+	// While we can add new topics matching extant regexps, we can't
+	// update that list on the fly.  We compile them once at startup.
+	// Changing them is a configuration change and requires a restart.
+
+	if len(k.TopicRegexps) == 0 {
+		// Nothing to do.
+		return
+	}
+	k.regexps = make([]regexp.Regexp, 0, len(k.TopicRegexps))
+	// Let's optimistically assume that all of them compile
+	for _, r := range k.TopicRegexps {
+		re, err := regexp.Compile(r)
+		if err == nil {
+			k.regexps = append(k.regexps, *re)
+			k.Log.Infof("Added regular expression '%s' to topics", r)
+		} else {
+			k.Log.Errorf("Regular expression '%s' did not compile: '%w'; not adding to topics", r, err)
+		}
+	}
+}
+
+func (k *KafkaConsumer) refreshTopics() error {
+	// We have instantiated a new generic Kafka client, so we can ask
+	// it for all the topics it knows about.  Then we build
+	// regexps from our strings, loop over those, loop over the
+	// topics, and if we find a match, add that topic to
+	// out topic set, which then we turn back into a list at the end.
+
+	if len(k.regexps) == 0 {
+		return nil
+	}
+	if k.topicClient == nil {
+		err := k.createTopicScanner()
+		if err != nil {
+			return fmt.Errorf("could not create client to match topics to regexps: %w", err)
+		}
+	}
+
+	allDiscoveredTopics, err := k.topicClient.Topics()
+	if err != nil {
+		return err
+	}
+	k.Log.Infof("discovered topics: %v", allDiscoveredTopics)
```

Comment:
Can we please make this a debug function?
```suggestion
	k.Log.Debugf("discovered topics: %v", allDiscoveredTopics)
```